fix: GL Entries for AP/AR Summary

UrvashiKishnani opened the pull request:
SQL query modified to fetch only those GL Entries for Accounts Payable Summary and Accounts Receivable Summary reports where the corresponding payment entry is in submitted state.

Issue:
Currently the AP and AR Summary report calculates the advance amount from payment entries. However, the filter to check whether the payment entry is in submitted state is not present.

Steps to reproduce:
1. Create a payment entry for a certain supplier which is not linked to any payment invoice. This will add the amount paid in the payment entry to the advance payment column in the accounts payable report.
2. Cancel the above payment entry. The advance payment column in the accounts payable report still reflects this amount and the paid amount column is adjusted to tally the outstanding amount. However, these numbers do not make sense against the Accounts Payable report and the calculated paid amount is incorrect.

Closes #22846

UrvashiKishnani committed:
fix: GL Entries for AP/AR Summary

SQL query modified to fetch only those GL Entries for Accounts Payable Summary and Accounts Receivable Summary reports where the corresponding payment entry is in submitted state.

deepeshgarg007 reviewed (changes requested):
@UrvashiKishnani No need for joins. Just add a filter for is_cancelled = 0 like shown below
![image](https://user-images.githubusercontent.com/42651287/109517855-b5180000-7acf-11eb-9d24-0360f1e5d1e4.png)

UrvashiKishnani committed:
fix: GL Entries for AP/AR Summary without SQL join

SQL query modified to fetch only those GL Entries for Accounts Payable Summary and Accounts Receivable Summary reports where the corresponding payment entry is not in cancelled state.
